File: Q8 two sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] ans = new int[2];
        for (int i = 0; i < nums.length - 1; ++i) {
            int complement = target - nums[i];
            for (int j = i + 1; j < nums.length; ++j) {
                if (nums[j] == complement) {
                    ans[0] = i;
                    ans[1] = j;
                    return ans;
                }
            }
        }
        return new int[0]; // Return empty array if no pair found
    }
}

File: Q8 two sum.py
class Solution:
    def twoSum(self, nums, target):
        # Initialize a dictionary to store the complement and its index
        complement_map = {}

        # Traverse the list using indices
        for i in range(len(nums)):
            num = nums[i]
            complement = target - num
            if complement in complement_map:
                # If the complement is found in the map, return the indices
                return [complement_map[complement], i]
            # Otherwise, add the number and its index to the map
            complement_map[num] = i

        # Return an empty list if no pair is found
        return []
